test/test_controller.js

```diff
@@ -0,0 +1,364 @@
+            }).then(
+            function () {
+                return controller_instance.addBuyerToWhitelist(addAccount, {from: addAccount});
+            }).then(
+            function () {
+                assert.isFalse(true, "Expect access exception. The function is only for owner");
+            }).catch(
+            function (error) {
+                return controller_instance.removeBuyerFromWhitelist(addAccount);
+            }).then(
+            function () {
+                return controller_instance.isAddressWhitelisted.call(addAccount);
+            }).then(
+            function (isAddrWhite) {
+                assert.isFalse(isAddrWhite.valueOf(), "Address is whitelisted");
+            });
+    });
+
+    it("test addBuyers to whitelist", function () {
+        var addAccounts = [accounts[4], accounts[5]];
+        var controller_instance;
+        return Controller.deployed().then(
+            function (inst) {
+                controller_instance = inst;
+                return controller_instance.addBuyers(addAccounts, {from: addAccount});
+            }).then(
+            function () {
+                assert.isFalse(true, "Expect access exception. The function is only for owner");
+            }).catch(
+            function (error) {
+                return controller_instance.addBuyers(addAccounts);
+            }).then(
+            function () {
+                return controller_instance.isAddressWhitelisted.call(addAccounts[0]);
+            }).then(
+            function (isWhitelisted) {
+                assert.isTrue(isWhitelisted.valueOf(), "0 index address is not whitelisted");
+                return controller_instance.isAddressWhitelisted.call(addAccounts[1]);
+            }).then(
+            function (isWhitelisted) {
+                assert.isTrue(isWhitelisted.valueOf(), "1 index address is not whitelisted");
+            });
+    });
+
+    it("test removeBuyers from whitelist", function () {
+        var addAccounts = [accounts[6], accounts[7]];
+        var controller_instance;
+        return Controller.deployed().then(
+            function (inst) {
+                controller_instance = inst;
+                return controller_instance.addBuyers(addAccounts);
+            }).then(
+            function () {
+                return controller_instance.removeBuyers(addAccounts);
+            }).then(
+            function () {
+                return controller_instance.isAddressWhitelisted.call(addAccounts[0]);
+            }).then(
+            function (isWhitelisted) {
+                assert.isFalse(isWhitelisted.valueOf(), "0 index address is whitelisted");
+                return controller_instance.isAddressWhitelisted.call(addAccounts[1]);
+            }).then(
+            function (isWhitelisted) {
+                assert.isFalse(isWhitelisted.valueOf(), "1 index address is whitelisted");
+            });
+    });
+});
+
+contract('ICO Controller', function (accounts) {
```

Comment:
In my opinion, if you split tests into blocks, the names of the blocks should be different.